### ISSUE TITLE
Is it possible that --exiftool runs even with --dry-run active?

### ISSUE THREAD

**Issue by oPromessa** [RhetTbull/osxphotos#641]
**Describe the bug**
I noticed that while running `osxphotos export` with `--update` on a previously exported folder; where I deleted one file for testing. But running with `--dry-run` it seems `exiftool` attempts to perform an update on destination file.

**To Reproduce**
1. What' the full command line you used with osxphotos?
   * TOML file
```
[export]
#cleanup = true
db = "/Users/Shared/Pictures/iPhoto Shared Library.photoslibrary"
description_template = "Album:{album,}{newline}Description:{descr,}{newline}Person:{person,}"
directory = "{album[/,.|:,.]}"
dry_run = true
exiftool = true
exiftool_merge_keywords = true
exiftool_merge_persons = true
from_date = 2021-12-20T00:00:00
keyword_template = [ "{keyword}", "{label}", "{searchinfo.activity?activity:{searchinfo.activity},}","{searchinfo.venue_type?venue_type:{searchinfo.venue_type},}", ]
not_hidden = true
report = "/Users/Shared/Pictures/Logs/colibri.csv"
retry = 2
skip_original_if_edited = true
strip = true
touch_file = true
update = true
verbose = true
```
2. What was the error output?
```
2022-02-23 21:12:01.662887 -- Loaded options from file /Users/MSP/Pictures/Pictures/Apps/colibri.toml
2022-02-23 21:12:01.662960 -- osxphotos version 0.46.1
2022-02-23 21:12:01.663749 -- exiftool path: /usr/local/bin/exiftool
Warning: export database '/Users/Shared/Pictures/Logs/.osxphotos_export.db' is in a different directory than export destination '/Volumes/photo-1/Family.Photos'
2022-02-23 21:12:52.377013 -- Using export database /Users/MSP/Pictures/Pictures/Logs/.osxphotos_export.db
2022-02-23 21:12:52.377105 -- Upgraded export database /Users/MSP/Pictures/Pictures/Logs/.osxphotos_export.db from version 4.3 to 6.0
2022-02-23 21:12:52,379 - WARNING - photosdb.py - 110 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 12.0, 12.1]: you have Darwin, OS version: 12.2
2022-02-23 21:12:52.381352 -- Processing database /Users/Shared/Pictures/iPhoto Shared Library.photoslibrary/database/photos.db
2022-02-23 21:12:52.384790 -- Processing database /Users/Shared/Pictures/iPhoto Shared Library.photoslibrary/database/Photos.sqlite
2022-02-23 21:12:52.384853 -- Database locked, creating temporary copy.
(...)
2022-02-23 21:13:29.635211 -- Exporting IMG_20211224_223711.JPG (IMG_20211224_223711.JPG) as IMG_20211224_223711.JPG (53/105)
2022-02-23 21:13:29.635553 -- Writing metadata with exiftool for IMG_20211224_223711.JPG
Error exporting photo (4F0AFFB9-BA9B-489B-9970-1D3612E9D014: IMG_20211224_223711.JPG) as IMG_20211224_223711.JPG: [Errno 2] No such file or directory: '/Volumes/photo-1/(...)/IMG_20211224_223711.JPG'
Retrying export for photo (4F0AFFB9-BA9B-489B-9970-1D3612E9D014: IMG_20211224_223711.JPG)
2022-02-23 21:13:29.636103 -- Writing metadata with exiftool for IMG_20211224_223711.JPG
Error exporting photo (4F0AFFB9-BA9B-489B-9970-1D3612E9D014: IMG_20211224_223711.JPG) as IMG_20211224_223711.JPG: [Errno 2] No such file or directory: '/Volumes/(...)/IMG_20211224_223711.JPG'
Retrying export for photo (4F0AFFB9-BA9B-489B-9970-1D3612E9D014: IMG_20211224_223711.JPG)
2022-02-23 21:13:29.636644 -- Writing metadata with exiftool for IMG_20211224_223711.JPG
Error exporting photo (4F0AFFB9-BA9B-489B-9970-1D3612E9D014: IMG_20211224_223711.JPG) as IMG_20211224_223711.JPG: [Errno 2] No such file or directory: '/Volumes/(...)/IMG_20211224_223711.JPG'
2022-02-23 21:13:29.637514 -- Exporting IMG_20211224_223701.JPG (IMG_20211224_223701.JPG) as IMG_20211224_223701.JPG (54/105)
2022-02-23 21:13:29.639487 -- Writing metadata with exiftool for IMG_20211224_223701.JPG
2022-02-23 21:13:29.798838 -- Exported updated file /Volumes/(...)MG_20211224_223701.JPG
(...)
```

**Expected behavior**
Would expect exiftool not to be run with `dry-run` option.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: MacOS 12.2.1
 - osxphotos version (`osxphotos --version`): 0.46.1

**Additional context**
`.osxphotos_db` file was probably updated due to the most recent changes in `osxphotos`.

**Comment by RhetTbull** [comment]
If this is happening, it shouldn't be. --dry-run should make absolutely no changes to the file system.

I would have though this code was blocked by this test -- will need to investigate why it's happening.

https://github.com/RhetTbull/osxphotos/blob/9964fd063522dea261bb0066a4afa18ee2751fb2/osxphotos/photoexporter.py#L1306-L1316

**Comment by RhetTbull** [comment]
It's not exiftool, it's the --touch-file code that's trying to do a stat on non-existent files.  Working on a fix. (But no files were actually getting touched with --dry-run)

**Comment by RhetTbull** [comment]
I think this is fixed in 0.46.4 which I'm about to release shortly.  The code was trying to stat a file to see if touch times needed to be updated but the file wasn't there (because it was running in dry run mode).  Came from a recent refactoring of the export logic.

**Comment by RhetTbull** [comment]
Fixed in [v0.46.4](https://github.com/RhetTbull/osxphotos/releases/tag/v0.46.4)
